### PR TITLE
Add codespace container for reproducing issues

### LINKED ISSUE #2753: Repro template
Many big projects offer repro templates that spin up an environment with everything needed make a minimal reproducible example simply by clicking a link.

Having something like this for `pymatgen` would make it much easier for users (esp. newcomers) to point out problems and reduce workload on maintainers if they get high-quality reproductions.

We could try out [GitHub's new Codespaces](https://docs.github.com/en/codespaces) for this.

Here's what that might look like:

Screenshot 2022-11-21 at 14 10 54

We'd link this codespace in our issue template and invite people to launch it, copy-paste their code into it and then paste the link to the codespace into their issue. (This would also ensure the error occurs on `master` and isn't due to outdated versions.

All we'd need to do is create a `.devcontainer/devcontainer.json` file that tells GitHub how to spin up our codespace and create a file like `test_files/repro-template.ipynb` with some common imports and maybe a dummy structure or simple `MPRester` query to get people up and running.

```json5
// .devcontainer/devcontainer.json
{
  "image": "mcr.microsoft.com/devcontainers/universal:2",
  "hostRequirements": {
    "cpus": 4
  },
  "waitFor": "onCreateCommand",
  "updateContentCommand": "python -m pip install -r requirements.txt && python -m pip install -e .",
  "customizations": {
    "codespaces": {
      "openFiles": ["test_files/repro_tempalte.ipynb"]
    },
    "vscode": {
      "extensions": [
        "ms-toolsai.jupyter",
        "ms-python.python"
      ]
    }
  }
}
```

### PR DESCRIPTION
**Pull request by DanielYang59**
Tasks finished to resolve #2753:
- Added a codespace container configuration file.
- Added a brief guide to use codespace container for issue reproduction.
- Added link in issue template to encourage users to try this new workspace.

However I still need advice on handling user feedbacks, as I failed to identify any way to create a link or share a snapshot to an existing codespace, as @janosh suggested.

> We'd link this codespace in our issue template and invite people to launch it, copy-paste their code into it and then paste the link to the codespace into their issue.

The closest I could find currently is the [Live Share](https://docs.github.com/en/codespaces/developing-in-a-codespace/working-collaboratively-in-a-codespace), which allows multiple users to work on a single codespace in real-time. But I don't think this is what we're looking for though. This would require the user to publicly share a link to a codespace and might have security risks.

So far the best solution to me still remains to run the code in the codespace, and **paste the output back to the issue** manually (until a feature such as sharing a snapshot is out). In doing so, we could help pre-configure the reproduction environment and information collection mechanisms, also avoiding all sorts of non-pymatgen issues (python not installed properly, outdated pymatgen or dependency, and so on).

### CODE REVIEW

File: .devcontainer/README.md
Comment (janosh) [inline .devcontainer/README.md:1]: Shouldn't this markdown be part of the issue template?

File: .devcontainer/create_codespace.png
Comment (janosh) [inline .devcontainer/create_codespace.png:1]: Nice, very helpful for users. We don't need to check this file in though. Could you compress it, then upload it to a github issue? just drag and drop the file into the markdown area, no need to actually open the issue. GH will give you a link to the file on their servers which you can use anywhere.

**Review by janosh — CHANGES_REQUESTED** [review]

**Comment by janosh** [comment]
laos, could you rebase your branch onto pymatgen `master`?

**Comment by DanielYang59** [comment]
I'm really sorry @janosh. I must have been messing with commits trying to rebase and combine my commit history. Should I close this PR and reopen a new one?

File: .devcontainer/create_codespace.png
Comment (DanielYang59) [reply .devcontainer/create_codespace.png:1]: Thanks. New hack learned.

**Pull request closed without merging**